changelog/371.feature.rst

```diff
@@ -0,0 +1,2 @@
+Update ``Client`` classes such that their initialization kwargs are explicitly stated and typehinted.
+- replaced ``**kwargs`` / ``**options`` with explicit keyword arguments for the ``__init__`` methods of ``Client``, ``Bot``, ``InteractionBot``, all ``AutoSharded*`` variants, and all relevant parent classes.
```

Comment:
```suggestion
Update :class:`Client` classes such that their initialization kwargs are explicitly stated and typehinted.
- Replaced ``**kwargs`` / ``**options`` with explicit keyword arguments for the ``__init__`` methods of :class:`Client`, :class:`ext.commands.Bot`, :class:`ext.commands.InteractionBot`, all ``AutoSharded*`` variants, and all relevant parent classes.
```

Additionally, should this be marked as a breaking change?

Reply:
Probably, yeah. How would I go about that regarding towncrier? Do I just change the filename to `371.breaking.rst`?

Reply:
Yes

Reply:
Done in 61fb43b455f0dc241d4ddce2a037d9878a349fc3.